scripts/gen_coverage.py

```diff
@@ -14,80 +14,218 @@
+    build_extension,
+    clean_build_dir,
+    err,
+    get_output,
+    install_editable,
+    make_cmake_args,
+    resolve_compilers,
+    run,
+    warn,
+)
+
+
+def find_bin_llvm(compiler):
+    if os.path.isabs(compiler):
+        bin_dir = os.path.dirname(compiler)
+    else:
+        compiler_path = get_output(["which", compiler])
+        if not compiler_path:
+            raise RuntimeError(f"Compiler {compiler} not found in PATH")
+        bin_dir = os.path.dirname(compiler_path)
+    compiler_dir = os.path.join(bin_dir, "compiler")
+    if os.path.exists(compiler_dir):
+        bin_llvm = compiler_dir
     else:
-        assert False, sys.platform + " not supported"
+        bin_dir = os.path.dirname(bin_dir)
+        bin_llvm = os.path.join(bin_dir, "bin-llvm")
+    if not os.path.exists(bin_llvm):
+        raise RuntimeError(f"--bin-llvm value {bin_llvm} not found")
+    return bin_llvm
 
-    if not IS_LIN:
-        raise RuntimeError(
-            "This scripts only supports coverage collection on Linux"
-        )
+
+def parse_args():
+    p = argparse.ArgumentParser(description="Build dpctl and generate coverage")
+
+    p.add_argument(
+        "--c-compiler", default=None, help="Path or name of C compiler"
+    )
+    p.add_argument(
+        "--cxx-compiler", default=None, help="Path or name of C++ compiler"
+    )
+    p.add_argument(
+        "--compiler-root",
+        type=str,
+        default=None,
+        help="Path to compiler installation root",
+    )
+    p.add_argument(
+        "--oneapi",
+        dest="oneapi",
+        action="store_true",
+        help="Use default oneAPI compiler layout",
+    )
+
+    p.add_argument(
+        "--verbose",
+        dest="verbose",
+        action="store_true",
+        help="Enable verbose makefile output",
+    )
+
+    p.add_argument(
+        "--no-level-zero",
+        dest="no_level_zero",
+        action="store_true",
+        default=False,
+        help="Disable Level Zero backend (deprecated: use --target-level-zero "
+        "OFF)",
+    )
+    p.add_argument(
+        "--target-level-zero",
+        action="store_true",
+        help="Enable Level Zero backend explicitly",
+    )
+
+    p.add_argument(
+        "--generator", type=str, default="Ninja", help="CMake generator"
+    )
+    p.add_argument(
+        "--cmake-executable",
+        type=str,
+        default=None,
+        help="Path to CMake executable used by build",
+    )
+
+    p.add_argument(
+        "--cmake-opts",
+        type=str,
+        default="",
+        help="Additional options to pass directly to CMake",
+    )
+
+    p.add_argument(
+        "--gtest-config",
+        help="Path to GTestConfig.cmake file for a custom GTest installation",
+    )
+    p.add_argument(
+        "--bin-llvm",
+        help="Path to folder where llvm-cov/llvm-profdata can be found",
+    )
+    p.add_argument("--skip-pytest", dest="run_pytest", action="store_false")
```

Comment:
Based on PR's description it seems it has to be `--no-pytest`, or the description in PR needs to be updated.